src/crimson/os/seastore/object_data_handler.h

```diff
@@ -77,8 +118,51 @@ struct ObjectDataBlock : crimson::os::seastore::LogicalCachedExtent {
   }
 
   void logical_on_delta_write() final {
+    if (overwrites) {
+      overwrites->copy_changes(get_bptr());
+      overwrites->clear();
+      overwrites = nullptr;
+    }
```

Comment:
Should this be moved to prepare_commit() which is at the prepare_record() time ?